src/System.Collections/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -247,9 +247,9 @@ private void CopyTo(KeyValuePair<TKey, TValue>[] array, int index)
                 throw new ArgumentNullException("array");
             }
 
-            if (index < 0 || index > array.Length)
+            if ((uint)index > (uint)array.Length)
```

Comment:
I get that it means the same thing, but it's very clear to me when I see the other comparisons exactly what's being checked for, whereas this takes a bit longer to think it through.  I'd prefer the original. (I'm not, however, suggesting that all of the existing occurrences of this pattern be changed.)